src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -160,6 +160,72 @@ abstract class RefChecks extends Transform {
         })
       }
     }
+    private def checkDubiousOverloads(clazz: Symbol): Unit = if (settings.warnDubiousOverload) {
+      // nilary members or methods with leading implicit params
+      def ofInterest(tp: Type): Boolean = tp match {
+        case mt: MethodType => mt.isImplicit
+        case PolyType(_, rt) => ofInterest(rt)
+        case _: NullaryMethodType | _ => true
+      }
+      def isNilary(tp: Type): Boolean = tp match {
```

Comment:
Didn't we call `foo` "nullary" and `foo()` "nilary"? So this should be `isNullary`?